### PR TITLE
API, Core - Change project to UWP C++

### PR DESCRIPTION
**Pull request by Take-A-Byte**
Move business logic to C++

### CODE REVIEW

File: ToDo.Core/TaskController.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                                         sender.GetHasCompleted());
+    break;
+  }
+}
+
+std::vector<std::reference_wrapper<API::IToDoTask>>
+TaskController::GetAlltasks() {
+  auto getAllTasks = std::async(
+      std::launch::async,
+      [&]() -> std::vector<std::reference_wrapper<API::IToDoTask>> {
```
Comment (sparkskapil) [inline ToDo.Core/TaskController.cpp]: In newer C++ the return-type for a lambda is optional.

File: ToDo.Core/TaskController.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          task.RegisterForPropertyChanged(
+              [&](ToDoTask &sender, TaskAttribute changedAttribute) {
+                OnTaskAttributesChanged(sender, changedAttribute);
+              });
+        }
+
+        return std::vector<std::reference_wrapper<API::IToDoTask>>();
+      });
+
+  return getAllTasks.get();
```
Comment (sparkskapil) [inline ToDo.Core/TaskController.cpp]: I think spawning a thread and joining in a same function defeats the purpose of a thread. 
Can we do it without it?

**Review by sparkskapil — COMMENTED** [review]
Overall design looks good. 
I think there can be a better approach for handling events.

File: ToDo.Core/TaskController.cpp (hunk 2 above)
Comment (Take-A-Byte) [reply ToDo.Core/TaskController.cpp]: https://github.com/Take-A-Byte/ToDoApp/pull/1/commits/0df14db7965f5b544fae7c424c67d15feb3e48ab

File: ToDo.Core/TaskController.cpp (hunk 1 above)
Comment (Take-A-Byte) [reply ToDo.Core/TaskController.cpp]: https://github.com/Take-A-Byte/ToDoApp/pull/1/commits/0df14db7965f5b544fae7c424c67d15feb3e48ab

**Pull request closed without merging**